service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomServiceImplTest.java

```diff
@@ -0,0 +1,299 @@
+                if (m.getName().equals(name)) return m;
+            }
+            return null;
+        }
+
+    }
+
+    MockDatabase database;
+    Mushroom mushroom1, mushroom2, mushroom3;
+
+    private static Mushroom setupMushroom(String name, MushroomType type, String fromM, String toM){
+        Mushroom m = new Mushroom();
+        m.setName(name);
+        m.setIntervalOfOccurrence(fromM,toM);
+        m.setType(type);
+        return m;
+    }
+
+    @Before
+    public void Setup(){
+        //TODO: actual setup
+
+        database = new MockDatabase();
+        mushroom1 = setupMushroom("some",MushroomType.UNEDIBLE,"june","july");
+        mushroom2 = setupMushroom("other",MushroomType.POISONOUS,"june","july");
+        mushroom3 = setupMushroom("different",MushroomType.UNEDIBLE,"may","september");
+
+
+
+        new Expectations(){{
+            mushroomDao.findById(anyLong);
+            result = new Delegate() {
+                Mushroom foo(Long id){
+                    return database.findById(id);
+                }
+            }; minTimes = 0;
+
+            mushroomDao.findByMushroomType((MushroomType) any);
+            result = new Delegate() {
+                List<Mushroom> foo(MushroomType type){
+                    return database.findByMushroomType(type);
+                }
+            }; minTimes = 0;
+
+            mushroomDao.create((Mushroom) any);
+            result = new Delegate() {
+                void foo(Mushroom m){
+                    database.create(m);
+                }
+            }; minTimes = 0;
+
+            mushroomDao.update((Mushroom) any);
+            result = new Delegate() {
+                void foo(Mushroom m){
+                    database.update(m);
+                }
+            }; minTimes = 0;
+
+            mushroomDao.delete((Mushroom) any);
+            result = new Delegate() {
+                void foo(Mushroom m){
+                    database.delete(m);
+                }
+            }; minTimes = 0;
+
+            mushroomDao.findAll();
+            result = new Delegate() {
+                List<Mushroom> foo(){
+                    return database.findAll();
+                }
+            }; minTimes = 0;
+
+            mushroomDao.findByName(anyString);
+            result = new Delegate() {
+                Mushroom foo(String name){
+                    return database.findByName(name);
+                }
+            }; minTimes = 0;
+
+            mushroomDao.findByIntervalOfOccurrence(anyString, anyString);
+            result = new Delegate() {
+                List<Mushroom> foo(String fromMonth, String toMonth){
+                    return database.findByIntervalOfOccurrence(fromMonth,toMonth);
+                }
+            }; minTimes = 0;
+        }};
+    }
+
+    @Test
+    public void findAllMushrooms(){
+
+        assertThat(service.findAllMushrooms()).isEmpty();
+
+        database.create(mushroom1);
```

Comment:
dependent test

Reply:
not really... tested entity is service, database is expected to be reliable (is a mock object)

Reply:
oh, my bad - reviewing too fast..